lib/datastruct/kvhash.c

```diff
@@ -20,6 +21,15 @@ rehash(struct kvhash * H)
 	uint32_t * new_hashes;
 	size_t i, pos;
 
+	/* Sanity check. */
+	assert(H->nslots > 0);
+
+	/* Check for overflow. */
+	if (H->nslots >= SIZE_MAX / 2) {
```

Comment:
This should be `>` rather than `>=`.  Also, it can be an assertion; there's no way `H->nslots` could ever get too big for `new_nslots` to overflow, since the allocations of `new_pairs` and `new_hashes` would have failed in the previous call to `rehash` (since they're allocating values of more than 1 byte each).